Sources/ShopifyCheckoutKit/CheckoutBridge.swift

```diff
@@ -39,12 +39,16 @@ enum CheckoutBridge {
 
 	static func instrument(_ webView: WKWebView, _ instrumentation: InstrumentationPayload) {
 		if let payload = instrumentation.toBridgeEvent() {
-			// This is being wrapped in a timer as such due to a bug in loading of the event listeners on latest schema version. Will be removed once bug is fixed
-			let msg = "setTimeout(function() { window.MobileCheckoutSdk.dispatchMessage('instrumentation', \(payload));}, 1000 );"
-			webView.evaluateJavaScript(msg)
+			sendMessage(webView, messageName: "instrumentation", messageBody: payload)
 		}
 	}
 
+	static func sendMessage(_ webView: WKWebView, messageName: String, messageBody: String?) {
+		let dispatchMessageBody = messageBody != nil ? "'\(messageName)', \(messageBody!)" : "'\(messageName)'"
```

Comment:
How come you have to force unwrap `messageBody` here since you already have a nil check?

Reply:
xcode gives a warning without it

```
String interpolation produces a debug description for an optional value; did you mean to make this explicit?
```

I guess it needs a little help